File: assignments/a1/q7/Python/Balanced.py
#Balanced:top#
#Balanced:import#

# <balanced> ::= <parens> AT
class Balanced(_Start): #Balanced:class#

    className = "Balanced"
    ruleString = "<balanced> ::= <parens> AT"
    parens = None

    def __init__(parens):
        #Balanced:init#
        self.parens = parens

#Balanced#

File: assignments/a1/q7/Java/Balanced.java
//Balanced:top//
//Balanced:import//
import java.util.*;

// <balanced> ::= <parens> AT
public class Balanced extends _Start /*Balanced:class*/ {

    public static final String $className = "Balanced";
    public static final String $ruleString =
        "<balanced> ::= <parens> AT";

    public Parens parens;

    public Balanced(Parens parens) {
//Balanced:init//
        this.parens = parens;
    }

    public static Balanced parse(Scan scn$, Trace trace$) {
        if (trace$ != null)
            trace$ = trace$.nonterm("<balanced>", scn$.lno);
        Parens parens = Parens.parse(scn$, trace$);
        scn$.match(Token.Match.AT, trace$);
        return new Balanced(parens);
    }

//Balanced//
}
